lib/web_ui/lib/src/engine/canvaskit/canvaskit_api.dart

```diff
@@ -43,6 +43,9 @@ external set windowFlutterCanvasKit(CanvasKit? value);
 @JS('window.flutterCanvasKit')
 external CanvasKit? get windowFlutterCanvasKit;
 
+@JS('window.flutterCanvasKitLoaded')
+external JSPromise? get windowFlutterCanvasKitLoaded;
```

Comment:
Do we need both? `flutterCanvasKitLoaded` will eventually resolve to the same value as `windowFlutterCanvasKit`.

Can we remove the CanvasKit getter and just always await for the promise?

Reply:
I think as long as `loadEntrypoint` exists, we'll need both. The `windowFlutterCanvasKitLoaded` promise only exists when `load` is called. In the `loadEntrypoint` case, on hot restart we just want to reuse the `windowFlutterCanvasKit` that was already set by the loading code (that happened in Dart).

Once `loadEntrypoint` is removed, a bunch of this code can be simplified quite a bit (in fact, I think we can remove all the dart-side CanvasKit loading code completely at that point).